backend/corpora/lambdas/api/v1/collection_uuid/upload.py

```diff
@@ -0,0 +1,37 @@
+from flask import make_response
+
+from .....common.corpora_config import CorporaConfig
+from .....common.corpora_orm import CollectionVisibility
+from .....common import upload_sfn
+from .....common.utils.db_utils import db_session
+from .....common.entities import Collection, Dataset
+from .....common.utils import dropbox
+from .....common.utils.exceptions import ForbiddenHTTPException, InvalidParametersHTTPException, TooLargeHTTPException
+from .....common.utils.math_utils import GB
+
+
+@db_session
+def link(collection_uuid: str, body: dict, user: str):
+
+    # Verify Dropbox URL
+    url = body["url"]
+    if not dropbox.verify(url):
+        raise InvalidParametersHTTPException("The dropbox shared link is invalid.")
+
+    # Get file info
+    url = dropbox.get_download_url_from_shared_link(url)
+    resp = dropbox.get_file_info(url)
+    if resp["size"] > CorporaConfig().upload_max_file_size_gb * GB:
+        raise TooLargeHTTPException()
+    if resp["name"].rsplit(".")[-1].lower() not in CorporaConfig().upload_file_formats:
```

Comment:
May need to check the name.  What if name doesn't have a "." character.  This will thrown an exception.

Reply:
I take that back, a missing "." in name won't cause a problem.